api/maketdata.go

```diff
@@ -114,3 +115,42 @@ func getTickersHandler(storage storage.Market) func(c *gin.Context) {
 		ginutils.RenderSuccess(c, blockatlas.TickerResponse{Currency: md.Currency, Docs: tickers})
 	}
 }
+
+// @Summary Get charts data for a specific coin
+// @Id get_charts_data
+// @Description Get the charts data from an market and coin/token
+// @Accept json
+// @Produce json
+// @Tags charts
+// @Param coin query int true "coin id"
+// @Param token query string false "token id"
+// @Param currency query string false "the currency to show charts" default(USD)
+// @Success 200 {object} blockatlas.ChartData
+// @Router /v1/market/charts [get]
+func getChartsHandler(router gin.IRouter) {
+	var charts = marketdata.InitCharts()
+	router.GET("/market/charts", func(c *gin.Context) {
+		coinQuery := c.Query("coin")
+		coinId, err := strconv.Atoi(coinQuery)
+		if err != nil {
+			ginutils.RenderError(c, http.StatusInternalServerError, "Invalid coin")
+			return
+		}
+		token := c.Query("token")
+
+		days, err := strconv.Atoi(c.Query("days"))
+		if err != nil {
+			ginutils.RenderError(c, http.StatusInternalServerError, "Invalid days count")
+			return
+		}
+
+		currency := c.DefaultQuery("currency", blockatlas.DefaultCurrency)
```

Comment:
I think we should reject not supported currencies

Reply:
It is used for the case when you haven't passed `currency=USD` as parameter. Just for request simplification sake.